src/system_data.jl

```diff
@@ -549,22 +507,16 @@ end
 get_components_by_name(::Type{T}, data::SystemData, args...) where {T} =
     get_components_by_name(T, data.components, args...)
 
-#get_component_time_series(::Type{T}, data::SystemData, args...) where T =
-#    get_component_time_series(T, data.time_series, args...)
-#get_time_series(::Type{T}, data::SystemData, component, args...) where T = get_time_series(
-#    T, component, args...
-#)
-get_time_series_initial_times(data::SystemData) =
-    get_time_series_initial_times(data.components)
-get_time_series_initial_time(data::SystemData) =
-    get_time_series_initial_time(data.components)
-get_time_series_last_initial_time(data::SystemData) =
-    get_time_series_last_initial_time(data.components)
-get_time_series_horizon(data::SystemData) = get_time_series_horizon(data.time_series_params)
+generate_forecast_initial_times(data::SystemData) =
+    generate_forecast_initial_times(data.time_series_params)
+get_forecast_total_period(data::SystemData) =
+    get_forecast_total_period(data.time_series_params)
+get_forecast_count(data::SystemData) = get_forecast_count(data.time_series_params)
```

Comment:
This term is a bit misleading to me.  Would `get_forecast_window_count` be better?

Reply:
ok, we could change it